### DISCUSSION TITLE
Is there any way to know if the event MouseWheel was scrolled up or down?

### DISCUSSION THREAD

**Discussion by LaBatata101** [fltk-rs/fltk-rs#1030] [Q&A] [answered]
I want to know if the widget Scroll was scrolled up or down.
Couldn't find anything about it in the docs.

**Comment by MoAlyousef** [comment, marked as answer]
You can check app::event_dy() in an Event::MouseWheel, which would return a MouseWheel::Up, Down or None. Example usage from fltk-egui crate:
https://github.com/fltk-rs/fltk-egui/blob/8e79b66c21160d742b1afd2a6e87c34e7379b031/src/lib.rs#L275

**Reply by LaBatata101** [reply]
Instead of using the mouse wheel to scroll if I scroll clicking and holding in the scrollbar with the left mouse button and drag down or up, is possible to capture this event?

**Reply by MoAlyousef** [reply]
You would have to save the Scroll’s scrollbar’s original callback then call it again in set_callback;

```rust

use fltk::{prelude::*, *};
fn main() {
    let a = app::App::default();
    let mut win = window::Window::default().with_size(400, 300);
    let scroll = group::Scroll::default().size_of_parent();
    let _btn = button::Button::new(160, 500, 80, 40, "click");
    let mut scrollbar = scroll.scrollbar();
    scrollbar.set_callback({
        let mut cb = scrollbar.callback();
        move |_| {
            println!("print something, and also run the default callback");
            if let Some(cb) = cb.as_mut() {
                (*cb)();
            }
        }
    });
    win.end();
    win.show();
    a.run().unwrap();
}
```
